e2e_tests/tests/test_sdk.py

```diff
@@ -0,0 +1,166 @@
+import random
+import time
+
+import pytest
+
+import determined.common.api.errors
+from determined.common import yaml
+from determined.experimental import client as _client
+from tests import config as conf
+
+
+@pytest.fixture(scope="session")
+def client() -> _client.Determined:
+    """
+    Reduce logins by having one session-level fixture do the login.
+    """
+    return _client.Determined(conf.make_master_url())
+
+
+@pytest.mark.e2e_cpu
+def test_completed_experiment_and_checkpoint_lookups(client: _client.Determined) -> None:
+    with open(conf.fixtures_path("no_op/single-one-short-step.yaml")) as f:
+        config = yaml.safe_load(f)
+    exp = client.create_experiment(config, conf.fixtures_path("no_op"))
+    assert exp.wait() == _client.ExperimentState.COMPLETED
+
+    trials = exp.get_trials()
+    assert len(trials) == 1, trials
+    trial = trials[0]
+    assert client.get_trial(trial.id).id == trial.id
+
+    ckpt = trial.top_checkpoint()
+    assert trial.select_checkpoint(latest=True).uuid == ckpt.uuid
+    assert exp.top_checkpoint().uuid == ckpt.uuid
+    assert ckpt.uuid in (c.uuid for c in exp.top_n_checkpoints(100))
+    assert client.get_checkpoint(ckpt.uuid).uuid == ckpt.uuid
+
+
+def _make_live_experiment(client: _client.Determined) -> _client.ExperimentReference:
+    # Create an experiment that takes a long time to run
+    with open(conf.fixtures_path("no_op/single-very-many-long-steps.yaml")) as f:
+        config = yaml.safe_load(f)
+
+    exp = client.create_experiment(config, conf.fixtures_path("no_op"))
+    # Wait for a trial to actually start.
+    start = time.time()
+    deadline = start + 90
+    while True:
+        trials = exp.get_trials()
+        if trials:
+            break
+        assert time.time() < deadline, "experiment took too long to start trials"
+        time.sleep(0.1)
+
+    return exp
+
+
+@pytest.mark.e2e_cpu
+def test_experiment_manipulation(client: _client.Determined) -> None:
+    exp = _make_live_experiment(client)
+
+    exp.pause()
+    with pytest.raises(ValueError, match="Make sure the experiment is active"):
+        # Wait throws an error if the experiment is paused by a user.
+        exp.wait(interval=0.1)
+
+    exp.activate()
+
+    exp.cancel()
+    assert exp.wait() == _client.ExperimentState.CANCELED
+    exp.wait()
```

Comment:
is this additional .wait() after the assert necessary?

Reply:
nope, that's a carryover from some intermediate state of this code